data/restli-csharp/datagenerator/src/test/GeneratedDataTemplate/com/linkedin/restli/datagenerator/integration/NestedUnion.cs

```diff
@@ -0,0 +1,111 @@
+using System.Collections.Generic;
+using System;
+
+// DO NOT EDIT - THIS FILE IS GENERATED BY restli-csharp
+// Generated from com/linkedin/restli/datagenerator/integration/NestedUnion.pdsc
+
+namespace com.linkedin.restli.datagenerator.integration
+{
+  /// <summary>
+  /// Testing union nested inside Array or DataMap for C#.
+  /// Also testing reserved names.
+  /// </summary>
+  public class NestedUnion
+  {
+
+    // required
+    public IReadOnlyList<UnionInArray> unionInArray { get; }
+
+
+    public NestedUnion(Dictionary<string, object> data)
+    {
+      object value;
+      // Retrieve data for unionInArray
+      if (data.TryGetValue("unionInArray", out value))
+      {
+        List<Dictionary<string, object>> data0 = (List<Dictionary<string, object>>)value;
+        List<UnionInArray> result0 = new List<UnionInArray>();
+        foreach (Dictionary<string, object> data1 in data0)
+        {
+          UnionInArray result1;
+          result1 = new UnionInArray(data1);
+          result0.Add(result1);
+        }
+        unionInArray = result0;
+
+      }
+
+    }
+
+    public NestedUnion(NestedUnionBuilder builder)
+    {
+      // Retrieve data for unionInArray
+      if (builder.unionInArray != null)
+      {
+        unionInArray = (IReadOnlyList<UnionInArray>)builder.unionInArray;
+
+      }
+      else
+      {
+        throw new System.ArgumentNullException("Required field with no default must be included in builder: unionInArray");
+      }
+    }
+
+    public class UnionInArray
+    {
+      public string asString { get; }
+      public int? asInt { get; }
+      public Mode? dataMode { get; }
+    
+      public enum Mode
+      {
+        String,
+        Int,
+        NULL
```

Comment:
Why are we using different literals for enum unknown symbol? I saw that some files you are using `@unknown`, some you are using `NULL`, make it consistent.